client/client.go

```diff
@@ -218,12 +221,39 @@ func (cli *Client) ClientVersion() string {
 // UpdateClientVersion updates the version string associated with this
 // instance of the Client. This operation doesn't acquire a mutex.
 func (cli *Client) UpdateClientVersion(v string) {
+	// If no version specified, negotiate with the server
+	if v == "" {
+		cli.DowngradeClientVersion()
+	}
```

Comment:
We don't make any promises about not breaking the API on `client/` it's not being released as a versioned package.